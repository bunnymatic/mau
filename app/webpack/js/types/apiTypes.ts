/** models **/
interface OpenStudiosParticipantType {
  id: number,
  userId: number,
  openStudiosEventId: number
}

/** api request/response shapes **/
interface ArtistsOpenStudiosRegisterForOsResponse {
  success: boolean,
  participating: boolean,
  participant?: OpenStudiosParticipantType
}
