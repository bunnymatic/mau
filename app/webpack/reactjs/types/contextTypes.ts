import { ArtPiece } from "@models/art_piece.model";

export interface ArtPiecesContext {
  artPieces?: ArtPiece[];
}
