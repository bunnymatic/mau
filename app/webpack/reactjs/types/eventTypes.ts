export type ClickEventHandler = (event: MouseEvent) => void;
