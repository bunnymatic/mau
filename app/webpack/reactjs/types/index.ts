// Put all types in here until it gets unweildy.
// Then we can bust it into smaller files and import them here
// so consumers can stay the same.

export * from "./apiTypes";
export * from "./eventTypes";
export * from "./modelTypes";
export * from "./utilTypes";
