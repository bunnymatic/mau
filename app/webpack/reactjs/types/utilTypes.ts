import { DateTime } from "luxon";
import { ReactNode } from "react";
export type Nullable<T> = T | null;

export interface TimeSlot {
  start: DateTime;
  end: DateTime;
}

export type MauButtonStyle = "primary" | "secondary";
export type MauButtonStyleAttr = Record<MauButtonStyle, Boolean>;
export type ClassNames = string | Array<string> | Record<string, boolean>;
export type ChildrenProp = {
  children?: ReactNode | Array<ReactNode>;
};
