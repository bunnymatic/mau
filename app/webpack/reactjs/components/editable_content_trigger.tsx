import { routing } from "@js/services/routing.service";
import * as types from "@reactjs/types";
import React, { FC } from "react";

interface EditableContentTriggerProps extends types.CmsDocument {}

export const EditableContentTrigger: FC<EditableContentTriggerProps> = ({
  cmsid,
  page,
  section,
}) => {
  const path = cmsid
    ? routing.editCmsDocumentPath({ id: cmsid })
    : routing.newCmsDocumentPath({ page, section });
  return (
    <a className="editable-content-trigger" href={path} target="_blank">
      edit me
    </a>
  );
};
