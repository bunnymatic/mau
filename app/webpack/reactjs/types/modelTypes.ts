/** models **/
import { Nullable } from "./utilTypes";

export type IdType = number;

interface ActiveRecordModel {
  id: IdType;
}

export interface CmsDocument extends ActiveRecordModel {
  page: string;
  section: string;
  cmsid: number;
}

export interface Artist extends ActiveRecordModel {
  slug: string;
}

export interface Studio extends ActiveRecordModel {
  slug: string;
}

export interface ArtPiece extends ActiveRecordModel {
  title: string;
  dimensions: string;
  year: string;
  price: string;
}

export interface JsonApiModel<T, R> {
  id: string;
  type: string;
  attributes: T;
  relationships: R;
}

type ImageSizes = "small" | "medium" | "large" | "original";

// JsonAPI models look pretty different
interface ArtPieceAttributes {
  artist_name: string;
  favorites_count: number;
  price: Nullable<number>;
  display_price: Nullable<string>;
  year: Nullable<string>;
  dimensions: Nullable<string>;
  title: string;
  artist_id: number;
  image_urls: Record<ImageSizes, string>;
  sold_at: Nullable<string>;
}

interface ArtPieceRelationships {
  artist: Object;
  tags: Object;
  medium: Object;
}

export interface JsonApiArtPiece
  extends JsonApiModel<ArtPieceAttributes, ArtPieceRelationships> {}

export interface OpenStudiosParticipant extends ActiveRecordModel {
  userId: IdType;
  openStudiosEventId: IdType;
  showEmail: Nullable<boolean>;
  showPhoneNumber: Nullable<boolean>;
  shopUrl: Nullable<string>;
  videoConferenceUrl: Nullable<string>;
  videoConferenceSchedule: Nullable<Record<string, boolean>>;
  youtubeUrl: Nullable<string>;
}

// Note: this does *not* match the Ruby model OpenStudiosEvent but
// matches the response shape we use in `open_studios_status.html.slim` when we build
// the react component.  Not sure if this should be somewhere else but for now...

export interface OpenStudiosEvent {
  dateRange: string;
  startTime: string;
  endTime: string;
  specialEvent: SpecialEventDetails;
}

export interface DateInfo {
  startTime: string;
  endTime: string;
  startDate: string;
  endDate: string;
}

export interface SpecialEventDetails extends DateInfo {
  dateRange: string;
  timeSlots: string[];
}

export interface EmailAttributes {
  id: IdType;
  name?: string;
  email: string;
  createdAt: string;
  updatedAt: string;
}

interface GenericEvent {
  id: number;
  data?: Record<string, string | Record<string, string>>;

  createdAt: string;
  message: string;
  updatedAt: string;
}

interface OpenStudiosSignupEvent extends GenericEvent {}

interface UserChangedEvent extends GenericEvent {}

export interface ApplicationEvent {
  genericEvent?: GenericEvent;
  userChangedEvent?: UserChangedEvent;
  openStudiosSignupEvent?: OpenStudiosSignupEvent;
}
