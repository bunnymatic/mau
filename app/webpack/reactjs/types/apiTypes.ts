/** api request/response shapes **/

import { ApplicationEvent } from "./modelTypes";

export interface ContactArtistFormData {
  name: string;
  phone?: string;
  email?: string;
  message?: string;
  artPieceId: number;
}

export interface ApplicationEventsListResponse {
  applicationEvents: ApplicationEvent[];
}
