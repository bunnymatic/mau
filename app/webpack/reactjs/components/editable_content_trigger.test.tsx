import * as types from "@reactjs/types";
import { render, screen } from "@testing-library/react";
import React from "react";

import { EditableContentTrigger } from "./editable_content_trigger";

describe("EditableContentTrigger", () => {
  const renderComponent = (
    opts: types.CmsDocument = {
      cmsid: undefined,
      page: "myPage",
      section: "mySection",
    }
  ) => {
    return render(<EditableContentTrigger {...opts} />);
  };

  it("renders an edit button", () => {
    const { container } = renderComponent();
    expect(container).toMatchSnapshot();
  });

  it("given a cmsid, it makes an edit path", () => {
    renderComponent({ cmsid: 5 });
    const trigger = screen.queryByText("edit me");
    const link = trigger.closest("a");
    expect(trigger).toBeInTheDocument();
    expect(link.href).toEqual("http://localhost/admin/cms_documents/5/edit");
  });

  it("given no cmsid, it makes an new path", () => {
    renderComponent();
    const trigger = screen.queryByText("edit me");
    const link = trigger.closest("a");
    expect(trigger).toBeInTheDocument();
    expect(link.href).toEqual(
      "http://localhost/admin/cms_documents/new?cms_document[page]=myPage&cms_document[section]=mySection"
    );
  });
});
