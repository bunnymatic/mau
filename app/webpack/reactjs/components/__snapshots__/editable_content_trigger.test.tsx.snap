// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EditableContentTrigger renders an edit button 1`] = `
<div>
  <a
    class="editable-content-trigger"
    href="/admin/cms_documents/new?cms_document[page]=myPage&cms_document[section]=mySection"
    target="_blank"
  >
    edit me
  </a>
</div>
`;
